core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -1190,6 +1195,22 @@ private void createStageExecution(SubPlan subPlan, boolean rootFragment, Map<Sta
             }
         }
 
+        private boolean isNoMemoryFragment(PlanFragment fragment)
+        {
+            if (!fragment.getRemoteSourceNodes().stream().flatMap(node -> node.getSourceFragmentIds().stream())
+                    .allMatch(sourceFragmentId -> stageExecutions.get(getStageId(sourceFragmentId)).isNoMemoryFragment())) {
+                return false;
+            }
+
+            // If fragment source is not reading any external tables or only accesses information_schema assume it does not need significant amount of memory.
+            // Allow scheduling even if whole server memory is pre allocated.
```

Comment:
will we schedule unbounded number of information_schema queries? there must be some sort of a limit for them.

Reply:
If we do not have resource queues in place yes. But one can argue this is a mis-setup anyway - and very much possible in streaming mode right now.